src/gui/qgsgui.cpp

```diff
@@ -71,8 +72,14 @@ QgsLayoutItemGuiRegistry *QgsGui::layoutItemGuiRegistry()
   return instance()->mLayoutItemGuiRegistry;
 }
 
+void QgsGui::enableAutoPostionRestore( QWidget *widget, const QString &key )
+{
+  instance()->mWidgetStateHelper->registerWidget( widget, key );
+}
+
 QgsGui::~QgsGui()
 {
+  delete mWidgetStateHelper;
```

Comment:
I suspect this should be last deleted last for safety. Or better, parent it to the instance and don't manually delete it at all.

Reply:
Why is the other stuff not parented here?

Reply:
ahh because QgsGui isn't a QObject.  Will just delete later.

Reply:
Oh ok, that :P